### PR TITLE
Removed componentWillReceiveProps, react upgrade

### PR DESCRIPTION
**Pull request by wakooka**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Unreleased
 
-- [...]
+- **[UPDATE]** Upgraded React to 16.9.0
```
Comment (mcellier) [inline CHANGELOG.md]: Why not `16.10.1`? :-)

File: src/timePicker/TimePicker.tsx [hunk 2]
```diff
@@ -88,17 +88,14 @@ export default class TimePicker extends PureComponent<TimePickerProps, TimePicke
...
 
-  componentWillReceiveProps(newProps: TimePickerProps) {
-    const shouldRegenerateTimeSteps =
-      newProps.minuteStep !== this.props.minuteStep ||
-      newProps.timeStart !== this.props.timeStart
-
-    if (shouldRegenerateTimeSteps) {
-      this.setState(state => ({
-        value: state.value < newProps.timeStart ? newProps.timeStart : state.value,
-        steps: this.generateTimeSteps(newProps),
```
Comment (pith) [inline src/timePicker/TimePicker.tsx]: So the steps can no more be updated ? That's a breaking change ?

Comment (pith) [reply src/timePicker/TimePicker.tsx]: Not sure it should be in the state however.

File: package.json [hunk 3]
```diff
@@ -40,13 +40,13 @@
...
     "@babel/preset-react": "7.0.0",
     "@blablacar/cli": "1.2.0",
-    "@storybook/addon-a11y": "5.2.1",
-    "@storybook/addon-actions": "5.2.1",
-    "@storybook/addon-knobs": "5.2.1",
-    "@storybook/addon-options": "5.2.1",
-    "@storybook/addon-viewport": "5.2.1",
-    "@storybook/addons": "5.2.1",
-    "@storybook/react": "5.2.1",
+    "@storybook/addon-a11y": "5.0.3",
```
[the remark this answers is not in the document]
Comment (mcellier) [reply package.json]: It's written in the changelog: The newest version of storybook has some issues with knobs (we can change values only once)

File: src/textField/TextField.tsx [hunk 4]
```diff
@@ -119,13 +117,19 @@ export default class TextField extends PureComponent<TextFieldProps, TextFieldSt
     }
   }
 
-  componentWillReceiveProps({ defaultValue, focus }: TextFieldProps) {
-    if (this.props.defaultValue !== defaultValue) {
-      this.setState({
-        value: defaultValue,
-        previousValue: this.state.value,
-      })
+  static getDerivedStateFromProps(props: TextFieldProps, state: TextFieldState) {
```
[the remark this answers is not in the document]
Comment (mcellier) [reply src/textField/TextField.tsx]: That's what we saw, and we thought about changing the behavior, but we're not sure we'll not break anything in kairos.

File: src/timePicker/TimePicker.tsx (hunk 2 above)
Comment (mcellier) [reply src/timePicker/TimePicker.tsx]: It was not working and not used. This is a really edge case.

File: src/autoComplete/AutoCompleteList.tsx [hunk 5]
```diff
@@ -41,18 +42,19 @@ export default class AutoCompleteList extends Component<
...
   }
 
-  componentWillReceiveProps(nextProps: AutoCompleteListProps) {
-    this.setState({
-      highlightedIndex: null,
-    })
-
-    if (canUseEventListeners && nextProps.visible !== this.props.visible) {
-      if (nextProps.visible) {
+  componentDidUpdate({ visible, items }: AutoCompleteListProps) {
```
[the remark this answers is not in the document]
Comment (wakooka) [reply src/autoComplete/AutoCompleteList.tsx]: I agree, I changed all componentDidUpdate in the codebase to use `prevProps`

**Comment by wakooka** [comment]
@pith PR updated

**Pull request merged**